Source/WebCore/css/ComputedStyleExtractor.cpp

```diff
@@ -574,6 +574,36 @@ static RefPtr<CSSValue> positionOffsetValue(const RenderStyle& style, CSSPropert
     return CSSPrimitiveValue::create(CSSValueAuto);
 }
 
+// FIXME: Replace calls to this function with `getCSSPropertyValuesForShorthandProperties`
+// once white-space longhands are completely implemented.
+Ref<CSSValue> ComputedStyleExtractor::getWhiteSpaceValue(const RenderStyle& style)
+{
+    auto whiteSpaceCollapse = style.whiteSpaceCollapse();
+    auto textWrap = style.textWrap();
+    if (whiteSpaceCollapse == WhiteSpaceCollapse::BreakSpaces && textWrap == TextWrap::Wrap)
+        return CSSPrimitiveValue::create(CSSValueBreakSpaces);
+    if (whiteSpaceCollapse == WhiteSpaceCollapse::Collapse && textWrap == TextWrap::Wrap)
+        return CSSPrimitiveValue::create(CSSValueNormal);
+    if (whiteSpaceCollapse == WhiteSpaceCollapse::Collapse && textWrap == TextWrap::NoWrap)
+        return CSSPrimitiveValue::create(CSSValueNowrap);
+    if (whiteSpaceCollapse == WhiteSpaceCollapse::Preserve && textWrap == TextWrap::NoWrap)
+        return CSSPrimitiveValue::create(CSSValuePre);
+    if (whiteSpaceCollapse == WhiteSpaceCollapse::PreserveBreaks && textWrap == TextWrap::Wrap)
+        return CSSPrimitiveValue::create(CSSValuePreLine);
+    if (whiteSpaceCollapse == WhiteSpaceCollapse::Preserve && textWrap == TextWrap::Wrap)
+        return CSSPrimitiveValue::create(CSSValuePreWrap);
+
+    // Reachable only if white-space-collapse/text-wrap properties were individually set
+    // instead of being changed via white-space shorthand keyword. If the appropriate
+    // feature flags for white-space are enabled, then fall back to separate longhands.
+    auto& settings = m_element->document().settings();
+    if (settings.cssWhiteSpaceCollapseEnabled() && settings.cssTextWrapEnabled())
+        return getCSSPropertyValuesForShorthandProperties(whiteSpaceShorthand());
+
+    // The white-space property cannot be properly expressed in terms of its longhands
+    return CSSPrimitiveValue::create(CSSValueInvalid);
```

Comment:
I would `return nullptr;` in this case (and have this function return a `RefPtr`)